### DISCUSSION TITLE
do ignore files ignore things based on file path or file contents?

### DISCUSSION THREAD

**Discussion by jzhao2007** [BurntSushi/ripgrep#2258] [Q&A] [answered]
#### What version of ripgrep are you using?

ripgrep 13.0.0 (rev af6b6c543b)

#### How did you install ripgrep?
unzip

#### What operating system are you using ripgrep on?
Windows 10

#### Describe your bug.
--ignore-file optional not working

#### What are the steps to reproduce the behavior?
echo aaa >> test.txt
echo aaa >> test2.txt
rg --ignore-file test.txt aaa

#### What is the actual behavior?

```
test2.txt
1:aaa

test.txt
1:aaa

```

#### What is the expected behavior?
-- test.txt should be excluded from the result

test2.txt
1:aaa

**Comment by BurntSushi** [comment, marked as answer]
Ignore files only ignore things based on the _file path_. From your example, it seems like you've somehow come under the impression that they work by looking at the _file contents_. They don't, just like `.gitignore` files don't. Only file paths. So if you want to ignore `test2.txt`, you need to put `test2.txt` in your ignore file.

I've converted this to a Discussion Question because this isn't a bug. And I made the title a bit more descriptive.

**Comment by jzhao2007** [comment]
Hi, thanks for quick response.
I get confused here. I am  using file path. There are two files containing
aaa text, test1.txt and test2.txt. when searching, I want to ignore one of
them test1.txt and only test2.txt is returned
So I am using
Rg --ignore-file test1.txt aaa
I did not specify the folder, the default is current folder.

**Reply by jzhao2007** [reply]
I think I understand what you talk about now. I need to create a separate file and put all files path I want to ignore in the file just like .gitignore.
This means, I can not do adhoc exclusion from command line, I must create a static exclusion file first.

**Reply by BurntSushi** [reply]
Of course you can do ad hoc exclusion on the command line. That's done via `-g '!test1.txt'`, for example. Have you read the [GUIDE](https://github.com/BurntSushi/ripgrep/blob/master/GUIDE.md)? It has a [whole section on manual filtering using globs](https://github.com/BurntSushi/ripgrep/blob/master/GUIDE.md#manual-filtering-globs).

**Comment by jzhao2007** [comment]
super. that is exactly what I am looking for.
I missed this -g option  obviously.

the command for my case should be
rg -g "test.txt" aaa


Thanks alot


On Thu, Jul 14, 2022 at 9:53 AM Andrew Gallant ***@***.***>
wrote:

> Of course you can do ad hoc exclusion on the command line. That's done via -g
> '!test1.txt', for example. Have you read the GUIDE
> <https://github.com/BurntSushi/ripgrep/blob/master/GUIDE.md#manual-filtering-globs>
> ?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/BurntSushi/ripgrep/discussions/2258#discussioncomment-3147883>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AD55YGXTPQ46G5ECU6YS3U3VUASYLANCNFSM53SJF44Q>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Thanks

Johnson Zhao

**Comment by jzhao2007** [comment]
by the way, great program , much faster than grep and ag.

**Reply by BurntSushi** [reply]
Thanks! :-)
